[jinxuan-owyong] iP

jinxuan-owyong opened the pull request.

jinxuan-owyong committed:
Refactor logo as method

jinxuan-owyong committed:
Add greet

jinxuan-owyong committed:
Update gitignore

jinxuan-owyong committed:
Implement echo

jinxuan-owyong committed:
Add indentation for output

jinxuan-owyong committed:
Implement add task

jinxuan-owyong committed:
Implement list tasks

jinxuan-owyong committed:
Implement task as class

jinxuan-owyong committed:
Implement mark/unmark task

jinxuan-owyong committed:
Refactor code for retrieving task details

Added describe() method which outputs whether the task is checked
and returns a string with the corresponding checkbox and task
name. Reduces amount of messy code in the main chat controller.

jinxuan-owyong committed:
Add child variants for Task

Added three types of tasks
1. ToDos: tasks without any date/time attached to it
2. Deadlines: tasks that need to be done before a specific date/
   time
3. Events: tasks that start at a specific date/time and ends at a
   specific date/time

jinxuan-owyong committed:
Remove magic numbers

Add State enum to keep track of program state

jinxuan-owyong committed:
Add support to create and list new tasks

Allows the user to create new todos, deadlines and events.
Validates user input using regex and checks whether the / keyword
is present.

Ng-YZ reviewed (commented):
Nicely done: Naming of code follows the Java coding standards closely, layout generally follows 4-spaces indentation, and 8-spaces indentation for wrapped text, K&R style used. It was very easy to read for most sections.

MingEn82 commented on src/main/java/ToDo.java:
```diff
@@ -1,12 +1,35 @@
+        }
+        result.add(input);
+        return result;
+    }
```
Code is easy to read, with good choice of method and variable names!

MingEn82 commented on src/main/java/Task.java:
```diff
@@ -31,4 +35,15 @@ public static String getCheckbox(boolean checked, String marker) {
+        result.add(input);
+        return result;
+    }
 }
```
Code is easy to read!

MingEn82 commented on src/main/java/Event.java:
```diff
@@ -1,16 +1,49 @@
+        result.add(input.substring(fromIndex + KEYWORD_FROM.length(), toIndex).trim());
+        result.add(input.substring(toIndex + KEYWORD_TO.length()).trim());
+        return result;
+    }
```
Overall well done! Code is easy to read with return statements are at the top of the methods.

MingEn82 commented on src/main/java/Duke.java:
```diff
@@ -111,6 +159,10 @@ public static void main(String[] args) {
+            }
         } while (true);
 
         scan.close();
```
Well done! Code is easy to read

MingEn82 commented on src/main/java/Deadline.java:
```diff
@@ -1,15 +1,44 @@
+        result.add(input.substring(0, byStart).trim());
+        result.add(input.substring(byStart + KEYWORD_BY.length()).trim());
+        return result;
+    }
```
Code is easy to read! A lot of effort have been made to stick to coding standards!

Ng-YZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+    static final String COMMAND_EXIT = "bye";
+    static final String COMMAND_LIST = "list";
+    static final String COMMAND_MARK = "mark";
+    static final String COMMAND_UNMARK = "unmark";
```
I like this section! Very clear and well-named to match coding standards!

Ng-YZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+        Scanner scan = new Scanner(s);
+        while (scan.hasNextLine()) {
+            System.out.println(INDENT + scan.nextLine());
+        }
```
Nicely indented relative to the conditionals

Ng-YZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+        for (int i = 0; i < numTasks; ++i) {
+            printWithIndentation((i + 1) + "." // number
+                    + "[" + (tasks[i].getIsCompleted() ? MARK : " ") + "] " // mark
+                    + tasks[i].getName()); // name
```
indentation done very clearly and according to standards!

Ng-YZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+        tasks[id].setIsCompleted(isCompleted);
+        if (isCompleted) { // mark task
+            printWithIndentation("Nice! I've marked this task as done:\n");
+        } else { // unmark task
```
maybe can consider putting the comments in a new line before the if-else statements?

Ng-YZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,116 @@
+
+    // data
+    static Task[] tasks = new Task[100];
+    static int numTasks = 0;
```
naming of variables follows convention, very easy to understand

jinxuan-owyong committed:
Improve code readability for command and status handling

Replaced if-else statements with switch-case statements.
Refactored command and status handling into methods using the
handleCommand<command> or handleStatus<status> rule.

jinxuan-owyong committed:
Move main method to top of Duke class

Improve logical flow of code by placing the main and processInput
method at the top of the Duke class

jinxuan-owyong committed:
Update output message for list and add task

jinxuan-owyong committed:
Fix regex pattern for command handling

Replace + with * to allow 1 character arguments

jinxuan-owyong committed:
Set testing javac output target to version 11

jinxuan-owyong committed:
Add test case

Added test case for basic task adding - todo, deadline, event

jinxuan-owyong committed:
Remove unused code

Removed unused imports and methods

jinxuan-owyong committed:
Improve clarity of naming and variable type

Named variables as constants as required. Added the private keyword
for variables which do not need public access. Renamed unclear
method names.

jinxuan-owyong committed:
Improve presentation of code

Added comments for unclear methods and restructured if-else
statements with ternary operators. Used line breaks to separate
chained methods.

jinxuan-owyong committed:
Implement error handling with try-catch

Replaced State with try-catch blocks and exceptions to allow for
central handling of exceptions.

jinxuan-owyong committed:
Add error handling for mark/unmark task

jinxuan-owyong committed:
Replace Task[] with ArrayList

jinxuan-owyong committed:
Merge branch 'branch-Level-5'

jinxuan-owyong committed:
Remove magic strings

jinxuan-owyong committed:
Organise classes into packages

jinxuan-owyong committed:
Update testing for packages

jinxuan-owyong committed:
Merge branch 'branch-A-Packages'

jinxuan-owyong committed:
Implement task deletion

jinxuan-owyong committed:
Implement data saving

Using the Gson library, tasks will be saved in data/save.txt.
The required directory will be created if not existing. Example
save data and test case has also been added.

jinxuan-owyong committed:
Merge branch 'branch-Level-6'

jinxuan-owyong committed:
Merge branch 'branch-Level-7'

slightlyharp commented on src/main/java/data/Data.java:
```diff
@@ -0,0 +1,98 @@
+import duke.tasks.ToDo;
+
+public class Data {
+    private static final String home = System.getProperty("user.dir");
```
Consider using all uppercase for constant

slightlyharp commented on src/main/java/data/Data.java:
```diff
@@ -0,0 +1,98 @@
+
+    private static Task deserializeJSON(String json) {
+        JsonObject j = gson.fromJson(json, JsonObject.class);
+        Task t;
```
Consider use a noun instead of single letter for naming of variable

slightlyharp commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,274 @@
+        String command = input.next().toLowerCase();
+
+        try {
+            switch (command) {
+            case COMMAND_LIST:
+                handleCommandList();
+                break;
+            case COMMAND_MARK:
+                handleCommandMark(input);
+                break;
+            case COMMAND_UNMARK:
+                handleCommandUnmark(input);
+                break;
+            case COMMAND_TODO:
+                handleCommandTodo(input);
+                break;
+            case COMMAND_DEADLINE:
+                handleCommandDeadline(input);
+                break;
+            case COMMAND_EVENT:
+                handleCommandEvent(input);
+                break;
+            case COMMAND_DELETE:
+                handleCommandDelete(input);
+            case COMMAND_SAVE:
+                handleCommandSave();
+                break;
```
Good job for consistent level of abstraction

slightlyharp commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,274 @@
+    private static final String CHAR_SPACE = " ";
+
+    // commands
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_SAVE = "save";
```
Good use of constant. You might want to put these into a separate class in the future when you do more OOP increment

slightlyharp commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+public class Deadline extends Task {
+    public static final String MARKER = "D";
+    private static final String KEYWORD_BY = "/by";
+    private final String by;
```
Consider use a noun for your variable

slightlyharp commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+            Pattern.CASE_INSENSITIVE);
+
+    public Deadline(ArrayList<String> details) {
+        super(details.get(0), TaskEnum.DEADLINE);
```
Good use of enum

slightlyharp reviewed (commented):
Very well done. Good use of constant and enum, methods are short and neat. Good example of code comments.

jinxuan-owyong committed:
Configure project for Maven

jinxuan-owyong committed:
Refactor Data as Storage

jinxuan-owyong committed:
Refactor outputs as class

jinxuan-owyong committed:
Refactor exceptions as individual classes

jinxuan-owyong committed:
Refactor task actions as class

jinxuan-owyong committed:
Refactor command handling as class

jinxuan-owyong committed:
Merge branch 'branch-A-MoreOOP'

jinxuan-owyong committed:
Refactor input parsing

jinxuan-owyong committed:
Cleanup code

Removed unused methods and imports

jinxuan-owyong committed:
Implement LocalDateTime for tasks

jinxuan-owyong committed:
Refactor Json parsing as class

jinxuan-owyong committed:
Implement search by keyword

jinxuan-owyong committed:
Refactor no task exception handling

jinxuan-owyong committed:
Add header comments to methods and classes

jinxuan-owyong committed:
Merge branch 'branch-Level-8'

jinxuan-owyong committed:
Merge branch 'branch-Level-9'

jinxuan-owyong committed:
Merge branch 'branch-A-JavaDoc'

jinxuan-owyong committed:
Merge branch 'master' into branch-A-JavaDoc

jinxuan-owyong committed:
Add header comments to methods and classes

jinxuan-owyong committed:
Merge branch 'branch-A-JavaDoc'

jinxuan-owyong committed:
Initialise Hugo site

jinxuan-owyong committed:
Add user guide

jinxuan-owyong committed:
Merge pull request #1 from jinxuan-owyong/branch-A-UserGuide

Add user guide

jinxuan-owyong committed:
Create hugo.yml

jinxuan-owyong committed:
Update hugo.yml

jinxuan-owyong committed:
Update docs

jinxuan-owyong committed:
Refactor task input parsing as class

jinxuan-owyong committed:
Reorder methods and variables alphabetically

jinxuan-owyong committed:
Redefine constant variables

jinxuan-owyong committed:
Refactor show search query results as method

jinxuan-owyong committed:
Refactor LocalDateTime (de)serialization as class

jinxuan-owyong committed:
Add corrupt data handling

jinxuan-owyong committed:
Reorganise parser structure

jinxuan-owyong committed:
Merge branch 'branch-EdgeCases'

jinxuan-owyong committed:
Fix UI when marking tasks

jinxuan-owyong committed:
Improve exit command

jinxuan-owyong committed:
Add help command

jinxuan-owyong committed:
Merge branch 'branch-A-UserGuide' of github.com:jinxuan-owyong/ip

jinxuan-owyong committed:
Update docs

jinxuan-owyong committed:
Merge branch 'branch-A-UserGuide'

jinxuan-owyong committed:
Remove wildcard imports

jinxuan-owyong committed:
Refactor boolean statements with lexical meaning

jinxuan-owyong committed:
Replace newline character with OS independent line separator

jinxuan-owyong committed:
Show help message on launch

jinxuan-owyong committed:
Add shutdown event handling

jinxuan-owyong committed:
Fix code spacing

jinxuan-owyong committed:
Rename methods and variables

jinxuan-owyong committed:
Refactor constants as class

jinxuan-owyong committed:
Fix exit message bug

jinxuan-owyong committed:
Merge branch 'branch-CodeImprovements'

jinxuan-owyong committed:
Update README.md

jinxuan-owyong committed:
Update version number

jinxuan-owyong committed:
Fix output formatting

jinxuan-owyong committed:
Fix read save data bug

jinxuan-owyong committed:
Handle event date time edge case
